### PR TITLE
Fix various Windows dependency issues

### LINKED ISSUE #23104: Windows build is broken with GStreamer 1.15
The package has apparently renamed `libffi-7.dll` to `ffi-7.dll`, so [this code](https://github.com/servo/servo/blob/7b8a898ac2ffef5ced8202dfd8bb40e37b972334/python/servo/build_commands.py#L622) fails. We should either tell developers to install a specific version of GStreamer or update it to support both old and newer versions.

### LINKED ISSUE #23125: "libgstgl-1.0-0.dll not found" when starting Servo Tech Demo on Windows 10
When starting Servo Tech Demo on Windows 10 the following error is thrown immediately:

```
servo.exe - System Error
The code execution cannot proceed because libgstgl-1.0-0.dll was not found. Reinstalling the program may fix the problem.
```

After closing the dialog, the browser does not start. This is the latest developper preview as served by download.servo.org today.

### PR DESCRIPTION
**Pull request by jdm**
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #23125 and fix #23104 
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23126)

### CODE REVIEW

**Comment by jdm** [comment]
I'm going to see what appveyor thinks of these changes.

**Comment by CYBAI** [comment]
Not sure if we need to change this line as well?

https://github.com/servo/servo/blob/dc55409757f4e61fb838d33a836747d6ed836e46/python/servo/build_commands.py#L614

Btw, not sure if it's better to detect gstreamer version and trim the `lib` prefix only for version higher than `1.15.2` ? 🤔

**Comment by jdm** [comment]
Good catch.

**Comment by UK992** [comment]
I have tested with both 1.15.1 and 1.15.2, and it works as expected.

File: python/servo/build_commands.py [hunk 1]
```diff
@@ -611,39 +611,52 @@ def build(self, target=None, release=False, dev=False, jobs=None,
...
+                    ["libintl-8.dll", "intl-8.dll"],
+                    ["liborc-0.4-0.dll", "orc-0.4-0.dll"],
+                    ["libwinpthread-1.dll", "winpthread-1.dll"],
+                    ["libz.dll", "libz-1.dll", "z-1.dll"]
                 ]
                 if gst_root:
                     for gst_lib in gst_dlls:
-                        shutil.copy(path.join(gst_root, "bin", gst_lib),
-                                    servo_exe_dir)
+                        if isinstance(gst_lib, str):
```
Comment (Manishearth) [inline python/servo/build_commands.py]: this pair of lines could be removed, but it's probably fine to keep them around

**Review by Manishearth — APPROVED** [review]
r=me

**Pull request merged**